aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs

```diff
@@ -30,22 +31,43 @@ macro_rules! convert_impl {
             move_storage_op: MoveStorageOp<Bytes>,
             legacy_creation_as_modification: bool,
         ) -> Result<WriteOp, VMStatus> {
+            let move_storage_op = match move_storage_op {
+                MoveStorageOp::New(data) => MoveStorageOp::New((data, None)),
+                MoveStorageOp::Modify(data) => MoveStorageOp::Modify((data, None)),
+                MoveStorageOp::Delete => MoveStorageOp::Delete,
+            };
             self.convert(
                 self.remote.$get_metadata_callback(state_key),
-                move_storage_op,
+                &move_storage_op,
```

Comment:
why &? We should consume here, and avoid clones in `convert`

Reply:
I am using move_storage_op again in the same function. So, had to use &.